cpp/daal/src/algorithms/kmeans/kmeans_plusplus_init_impl.i

```diff
@@ -247,25 +247,27 @@ template <typename algorithmFPType, CpuType cpu>
 class DataHelperCSR
 {
 public:
-    typedef BlockHelperCSR<algorithmFPType, cpu, CSRNumericTableIface> BlockHelperType;
+    typedef BlockHelperCSR<algorithmFPType, cpu, CSRNumericTable> BlockHelperType;
 
     DataHelperCSR(NumericTable * ntData)
-        : dim(ntData->getNumberOfColumns()), nRows(ntData->getNumberOfRows()), _nt(ntData), _csr(dynamic_cast<CSRNumericTableIface *>(ntData))
+        : dim(ntData->getNumberOfColumns()), nRows(ntData->getNumberOfRows()), _nt(ntData), _csr(dynamic_cast<CSRNumericTable *>(ntData))
     {}
     NumericTable * nt() const { return _nt; }
-    CSRNumericTableIface * ntIface() const { return _csr; }
+    CSRNumericTable * ntIface() const { return _csr; }
 
     Status updateMinDistInBlock(algorithmFPType * const minDistAccTrials, size_t nBlock, size_t iBlock, size_t nTrials, size_t iBestTrial,
                                 const algorithmFPType * aWeights, const algorithmFPType * const pLastAddedCenter, algorithmFPType * const aMinDist)
     {
         const size_t iStartRow      = iBlock * _nRowsInBlock;                                                  //start row
         const size_t nRowsToProcess = (iBlock == nBlock - 1) ? nRows - iBlock * _nRowsInBlock : _nRowsInBlock; //rows to process
 
-        ReadRowsCSR<algorithmFPType, cpu> ntDataBD(_csr, iStartRow, nRowsToProcess);
-        DAAL_CHECK_BLOCK_STATUS(ntDataBD);
-        const algorithmFPType * const pData = ntDataBD.values();
-        const size_t * const colIdx         = ntDataBD.cols();
-        const size_t * const rowIdx         = ntDataBD.rows();
+        // TODO: Better to use ReadRowsCSR, but there is a bug related to static library linking.
+        // Fixme when ReadRowsCSR will be fixed.
+        daal::data_management::CSRBlockDescriptor<algorithmFPType> block;
+        _csr->getSparseBlock(iStartRow, nRowsToProcess, daal::data_management::readOnly, block);
```

Comment:
`releaseSparseBlock(...)` must be used after `getSparseBlock(...)` to release the memory allocated with the `getSparseBlock`.
This version of the code produces memory leak.

Reply:
Done